pkg/services/alerting/conditions/query.go

```diff
@@ -333,11 +333,13 @@ func FrameToSeriesSlice(frame *data.Frame) (tsdb.TimeSeriesSlice, error) {
 		switch {
 		case field.Config != nil && field.Config.DisplayName != "":
 			ts.Name = field.Config.DisplayName
-		case field.Labels != nil:
+		case len(field.Labels) > 0:
 			ts.Tags = field.Labels.Copy()
 			// Tags are appended to the name so they are eventually included in EvalMatch's Metric property
 			// for display in notifications.
 			ts.Name = fmt.Sprintf("%v {%v}", field.Name, field.Labels.String())
+		case field.Config != nil && field.Config.DisplayNameFromDS != "":
```

Comment:
it needs to be above the labels case (otherwise having labels make DisplayNameFromDS have no effect